src/features/navbar/components/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,107 @@
+'use client';
+
+import './Navbar.scss';
+
+import { AppStrings as AppT } from '@core/dictionaries';
+import { Routes } from '@core/values';
+import { NavbarSize } from '@features/navbar';
+import { useNavbarSize } from '@features/navbar/hooks';
+import { NavbarConfig } from '@features/navbar/values/navbar-config';
+import { useAuthStore } from '@processes/auth';
+import { FontIcon } from '@shared/ui';
+import clsx from 'clsx';
+import { useRouter } from 'next/navigation';
+import { useState } from 'react';
+
+import NavbarLogo from '../NavbarLogo/NavbarLogo';
+import NavbarSizeControl from '../NavbarSizeControl/NavbarSizeControl';
+import NavbarVisibilityControl from '../NavbarVisibilityControl/NavbarVisibilityControl';
+
+const CATEGORIES = NavbarConfig.categories;
+const CATEGORIES_COUNT = CATEGORIES.length - 1;
+
+interface Props {
+  defaultSize?: NavbarSize;
+}
+
+function Navbar({ defaultSize }: Props) {
+  const router = useRouter();
+  const { navbarSize, setNavbarSize } = useNavbarSize(defaultSize);
+
+  const authorized = useAuthStore((state) => state.authorized);
+  const [isOpen, setVisibility] = useState(false);
+
+  const isShort = () => navbarSize === NavbarSize.SMALL;
```

Comment:
No need to use functions for such things, variable is perfectly fine: 
```typescript
const isShort = navbarSize === NavbarSize.SMALL;
```

Reply:
will the isShort be updated every time navbarSize is updated? The idea was to make an alias and take it out of the template

Reply:
Actually `isShort` value will be evaluated not only when `navbarSize` changes, but every time the component is rendered. For heave computations you might want to optimize it using `useMemo` hook, but here it's perfectly fine

Reply:
To summarize. `const isShort = navbarSize === NavbarSize.SMALL` will be updated:
- when is the redrawing taking place
- when are the props updated
- when navbarSize is updated
?

Do I understand correctly that we only do callbacks when we want to optimize using useMemo/useCallback?

Reply:
If you have a good article about the change detection mechanism in react, I would read it :)

Reply:
These are two great articles to better understand re-rendering in React:
- https://www.developerway.com/posts/react-re-renders-guide
- https://www.joshwcomeau.com/react/why-react-re-renders/

And you might also want to take a look useMemo/useCallback usages:
- https://www.developerway.com/posts/how-to-use-memo-use-callback
- https://www.joshwcomeau.com/react/usememo-and-usecallback/

Reply:
We do callback when we really need to use callback (e.g. to pass it down the tree and call from the children component or if it involves some computation that you want to keep all together)